drake/solvers/NPCPP/NonlinearProgram.cpp

```diff
@@ -0,0 +1,302 @@
+#include "NonlinearProgram.h"
```

Comment:
we're not going to have a nonlinear program class.  we want to build this into the optimizationproblem class as just a solve method.